lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -945,6 +1040,30 @@ bool OverrideMatcher::checkOverride(ValueDecl *baseDecl,
     emittedMatchError = true;
   }
 
+  auto baseGenericCtx = baseDecl->getAsGenericContext();
+  auto derivedGenericCtx = decl->getAsGenericContext();
+
+  // If the generic signatures are different, then complain.
+  auto newSig = getNewGenericSignature(baseDecl, decl);
+
+  if (newSig && derivedGenericCtx) {
```

Comment:
Is it possible for derivedGenericCtx == nullptr && newSig != nullptr? Maybe check if derivedGenericCtx and baseGenericCtx are both non-null before calling getNewGenericSignature()